src/main/java/com/starrocks/connector/spark/sql/StarRocksDataSourceProvider.java

```diff
@@ -40,74 +44,81 @@
 import java.util.Map;
 import javax.annotation.Nullable;
 
-import static com.starrocks.connector.spark.cfg.ConfigurationOptions.STARROCKS_FENODES;
-import static com.starrocks.connector.spark.cfg.ConfigurationOptions.STARROCKS_PASSWORD;
-import static com.starrocks.connector.spark.cfg.ConfigurationOptions.STARROCKS_USER;
-import static com.starrocks.connector.spark.sql.conf.StarRocksConfigBase.KEY_FE_HTTP;
+import static com.starrocks.connector.spark.cfg.ConfigurationOptions.STARROCKS_TABLE_IDENTIFIER;
+import static com.starrocks.connector.spark.cfg.ConfigurationOptions.makeWriteCompatibleWithRead;
+import static com.starrocks.connector.spark.sql.StarRocksTable.identifierFullName;
+import static com.starrocks.connector.spark.sql.conf.StarRocksConfig.PREFIX;
 
-public class StarRocksTableProvider implements RelationProvider, TableProvider, DataSourceRegister {
+public class StarRocksDataSourceProvider implements RelationProvider,
+        TableProvider,
+        DataSourceRegister {
 
-    private static final Logger LOG = LoggerFactory.getLogger(StarRocksTableProvider.class);
+    private static final Logger LOG = LoggerFactory.getLogger(StarRocksDataSourceProvider.class);
 
     @Nullable
-    private StarRocksSchema starocksSchema;
+    private StarRocksSchema starrocksSchema;
 
     @Override
     public BaseRelation createRelation(SQLContext sqlContext,
                                        scala.collection.immutable.Map<String, String> parameters) {
-        return new StarrocksRelation(sqlContext, Utils.params(parameters, LOG));
+        Map<String, String> mutableParams = new HashMap<>();
+        scala.collection.immutable.Map<String, String> transParameters = DialectUtils.params(parameters, LOG);
+        transParameters.toStream().foreach(key -> mutableParams.put(key._1, key._2));
+        SimpleStarRocksConfig config = new SimpleStarRocksConfig(makeWriteCompatibleWithRead(mutableParams));
+        starrocksSchema = getStarRocksSchema(config);
+
+        return new StarRocksRelation(sqlContext, transParameters);
     }
 
     @Override
     public StructType inferSchema(CaseInsensitiveStringMap options) {
         SimpleStarRocksConfig config = new SimpleStarRocksConfig(makeWriteCompatibleWithRead(options));
-        starocksSchema = getStarRocksSchema(config);
-        return InferSchema.inferSchema(starocksSchema, config);
+        starrocksSchema = getStarRocksSchema(config);
+        return InferSchema.inferSchema(starrocksSchema, config);
     }
 
     @Override
     public Table getTable(StructType schema, Transform[] partitioning, Map<String, String> properties) {
         SimpleStarRocksConfig config = new SimpleStarRocksConfig(makeWriteCompatibleWithRead(properties));
-        StarRocksSchema starRocksSchema = getStarRocksSchema(config);
-        return new StarRocksTable(schema, starRocksSchema, config);
+        starrocksSchema = getStarRocksSchema(config);
```

Comment:
Should not call `getStarRocksSchema` if `starrocksSchema` is not null? Otherwise `starrocksSchema` is meaningless